src/Wolverine/Transports/ListeningAgent.cs

```diff
@@ -227,7 +227,7 @@ public async ValueTask PauseAsync(TimeSpan pauseTime)
         _restarter = new Restarter(this, pauseTime);
     }
 
-    private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(0, 1);
+    private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
```

Comment:
This slipped in here for the issue i raised #1998 can make this a separate PR if thats agreed to use mutex vs signal pattern. But for now I wanted to ensure all tests pass